src/scikit_build_core/builder/get_requires.py

```diff
@@ -140,6 +156,9 @@ def dynamic_metadata(self) -> Generator[str, None, None]:
         if self.settings.fail:
             return
 
+        for build_require in self.settings.build.build_requires:
+            yield build_require.format(root=RootPath())
```

Comment:
I wanted to be more precise with the `RootPath` here, but not sure where to query the `pyproject.toml` path or equivalent. The other places seem to rely on being already in the appropriate path.

Reply:
I think it's required to be the cwd for a PEP 517 build.